controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,182 @@
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindingrequests/finalizers,verbs=update
+
+// Reconcile converts all the SpaceBindings created using the sandbox-cli to SpaceBindingRequests
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("reconciling SpaceBindingRequestMigration")
+
+	// Fetch the SpaceBinding instance
+	spaceBinding := &toolchainv1alpha1.SpaceBinding{}
+	err := r.Client.Get(ctx, request.NamespacedName, spaceBinding)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errs.Wrapf(err, "unable to get spacebinding")
+	}
+	if util.IsBeingDeleted(spaceBinding) {
+		logger.Info("the SpaceBinding is already being deleted")
+		return reconcile.Result{}, nil
+	}
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we can skip it
+	if hasSpaceBindingRequest(spaceBinding) {
+		return reconcile.Result{}, nil
+	}
+
+	spaceName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.Space}
+	space := &toolchainv1alpha1.Space{}
+	if err := r.Client.Get(ctx, spaceName, space); err != nil {
+		if errors.IsNotFound(err) {
+			// space was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading space
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
+	}
+
+	murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	if err := r.Client.Get(ctx, murName, mur); err != nil {
+		if errors.IsNotFound(err) {
+			// mur was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading MUR
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound MUR")
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
+		return ctrl.Result{}, errs.New("MasterUserRecord has no UserSignup owner reference")
+	}
+	// skip spaces with no creator label
+	if _, ok := space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey]; !ok {
+		// let's log an error, since this should not happen in production
+		logger.Error(errs.New("space has no SpaceCreatorLabelKey set"), "the spacebindings for this space will not be migrated", "space name", space.Name)
+		return ctrl.Result{}, nil
+	}
+	usersignup := mur.ObjectMeta.OwnerReferences[0]
```

Comment:
should we check if OwnerReference's kind is UserSignup?

Reply:
why not. Please see https://github.com/codeready-toolchain/host-operator/pull/922/commits/f0603c97e02c415d54f08701727abdab24b277a8

Reply:
I'm thinking we may look in the whole list of Owners to find the UserSignup entry. In this way we won't have the hard requirement that the UserSignup has to be the first of the list. WDYT?

something like the following:
```go
        // get the first UserSignup from the list of Owners
        // TODO: should we handle the scenario in which more than one UserSignup is set as owner?
	usersignup, err := func() (*metav1.OwnerReference, error) {
		for _, u := range mur.ObjectMeta.OwnerReferences {
			if u.Kind == "UserSignup" { // TODO: Group check?
				return &u, nil
			}
		}
		return nil, errs.New(fmt.Sprintf("owner reference of UserSignup not found for mur %s", murName))
	}()
	if err != nil {
		return ctrl.Result{}, err
	}
```

Other considerations on this piece of logic are:
- should we also check the group?
- should we handle the scenario in which more than one UserSignup is set as owner?

Reply:
This is a resource that is fully managed by our controllers, I don't think that we can rely on the assumption that there is always one owner-refence which is UserSignup

Reply:
@filariow I can create a lookup function for sure. I don't think we have to check the group and handle the case of multiple usersignups as reference. 

TBH I thought it would be safe to assume there is only one UserSignup as reference.

Reply:
@MatousJobanek since the resource is fully managed by our controllers , I thought instead that we could rely on that assumption. But I can implement the lookup as @filariow suggested. WDYT?

Reply:
you can use the lookup logic, but let's not over-engineer it and be pragmatic here. 
Btw, the second option of getting the UserSignup name is using the `.../owner` label from the MUR resource: 
https://github.com/codeready-toolchain/host-operator/blob/ad4c06f14f96db10a0ff794f413a362f62c7196e/controllers/usersignup/masteruserrecord.go#L39
so you don't have to struggle with the owner-references and the corner cases, and just pick the value of the label :wink:

Reply:
@MatousJobanek @filariow could you PTAL at https://github.com/codeready-toolchain/host-operator/pull/922/commits/7d0e0b9306cdb44dba68aa762af355bda822be50 and see if this looks good to you?

Reply:
Looks good, thanks :+1: